git-lfs.go

```diff
@@ -1,11 +1,37 @@
 package main
 
 import (
+	"fmt"
+	"os"
+	"os/signal"
+	"sync"
+
 	"github.com/github/git-lfs/commands"
 	"github.com/github/git-lfs/lfs"
 )
 
 func main() {
+	c := make(chan os.Signal)
+	signal.Notify(c)
+
+	var once sync.Once
+
+	go func() {
+		for {
+			sig := <-c
+			once.Do(lfs.ClearTempObjects)
```

Comment:
This clears temporary objects on *any* signal, including non-terminating signals like SIGSTOP, SIGCONT and SIGTRAP (the latter would kill the temp dir when a debugger was attached ;) ). Cleanup should only happen on default terminating signals, list here: https://en.wikipedia.org/wiki/Unix_signal#POSIX_signals